app/src/main/java/com/brandon/forzenbook/view/composables/CreateUserComposables.kt

```diff
@@ -46,104 +62,239 @@ fun CreateAccountContent(
+                    currentText = firstName,
+                    onTextChange = { firstName = it },
+                )
+                InputInfoTextField(
+                    hint = resources.getString(R.string.createAccountLastNameHint),
+                    currentText = lastName,
+                    onTextChange = { lastName = it },
+                )
+                ReadOnlyDateTextField(
+                    hint = resources.getString(R.string.createAccountBirthDateFormatHint),
+                    currentText = dateOfBirth,
+                    onTextChange = { dateOfBirth = it },
+                    characterLimit = DATE_OF_BIRTH_CHAR_LIMIT,
+                    onClickEvent = { dialog.show() }
+                )
+                if (state.isDateError) TextFieldErrorText(text = resources.getString(R.string.createAccountDateError))
+                InputInfoTextField(
+                    hint = resources.getString(R.string.createAccountEmailHint),
+                    onTextChange = { email = it },
+                    currentText = email,
+                    characterLimit = EMAIL_CHAR_LIMIT,
+                    onMaxCharacterLength = { emailError = it },
+                )
+                if (state.isEmailError) TextFieldErrorText(text = resources.getString(R.string.createAccountEmailError))
+                if (state.isDuplicateUser) TextFieldErrorText(text = resources.getString(R.string.createAccountDuplicateUserError))
+                InputInfoTextField(
+                    hint = resources.getString(R.string.createAccountLocationHint),
+                    onTextChange = { location = it },
+                    currentText = location,
+                    onMaxCharacterLength = { locationError = it },
+                    keyboardOptions = KeyboardOptions(imeAction = ImeAction.Go)
+                )
+                if (state.isLocationError) TextFieldErrorText(text = resources.getString(R.string.createAccountLocationError))
+                BlackButton(text = resources.getString(R.string.createAccountCreateButtonText)) {
+                    keyboardController?.hide()
+                    if (onCheckConnection()) {
+                        if (firstName.isNotEmpty() && lastName.isNotEmpty() && dateOfBirth.isNotEmpty() && email.isNotEmpty() && location.isNotEmpty()) {
+                            onSubmit(firstName, lastName, dateOfBirth, email, location)
+                        } else emptyFieldError = true
+                    } else isDialogOpen = true
+                }
+                if (emptyFieldError) TextFieldErrorText(text = resources.getString(R.string.requiredFieldsError))
+            }
         }
-        InputInfoTextField(
-            hint = resources.getString(R.string.createAccountLocationHint),
-            onTextChange = { location = it },
-            characterLimit = LOCATION_CHAR_LIMIT,
-            onMaxCharacterLength = { locationError = it },
+    )
+    if (isDialogOpen) {
```

Comment:
Make dialog launchings a LaunchedEffect

Reply:
Saw comment below, did make the change for the dialog that should be in a launched effect.